### PR TITLE
Add missing dartdocs from material.dart

### PR DESCRIPTION
**Pull request by abarth**
Making progress towards document all public APIs in material.dart. We're still
missing a few odds and ends (as well as missing docs in tabs.dart,
tooltip.dart, and two_level_list.dart).

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/material/button.dart [hunk 1]
```diff
@@ -50,6 +50,10 @@ class ButtonTheme extends InheritedWidget {
     Widget child
   }) : super(key: key, child: child);
 
+  /// Creates a button theme that is appropriate for footer buttons.
+  ///
+  /// For example, buttons at the bottom of [Dialog] or [Card] widgets use this
+  /// button theme.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/button.dart]: what's special about it?

File: packages/flutter/lib/src/material/data_table.dart [hunk 2]
```diff
@@ -742,7 +742,7 @@ class _SortArrowState extends State<_SortArrow> {
 
 }
 
-/* 
+/*
```
Comment (Hixie) [inline packages/flutter/lib/src/material/data_table.dart]: you can nuke this comment entirely, i have it on my branch here

File: packages/flutter/lib/src/material/button.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/material/button.dart]: It's denser and uses the accent color for the text.  /me will doc.

File: packages/flutter/lib/src/material/drawer.dart [hunk 3]
```diff
@@ -82,18 +82,28 @@ class Drawer extends StatelessWidget {
 /// * [Drawer]
 /// * [Scaffold.drawer]
 class DrawerController extends StatefulWidget {
+  /// Creates a controller for a [Drawer].
```
Comment (Hixie) [inline packages/flutter/lib/src/material/drawer.dart]: reading the class description and the constructor description I still don't know what a DrawerController is or why I would use one.

File: packages/flutter/lib/src/material/drawer.dart [hunk 4]
```diff
@@ -194,10 +204,12 @@ class DrawerControllerState extends State<DrawerController> {
     }
   }
 
+  /// Starts an animation to open the drawer.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/drawer.dart]: is this the expected entry-point for programmatically opening drawers? Or is there some method on Scaffold or something to do it?

File: packages/flutter/lib/src/material/drawer.dart [hunk 5]
```diff
@@ -194,10 +204,12 @@ class DrawerControllerState extends State<DrawerController> {
...
   void open() {
     _controller.fling(velocity: 1.0);
   }
 
+  /// Starts an animation to close the drawer.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/drawer.dart]: ditto

File: packages/flutter/lib/src/material/drop_down.dart [hunk 6]
```diff
@@ -218,12 +218,18 @@ class _DropDownRoute<T> extends PopupRoute<_DropDownRouteResult<T>> {
   }
 }
 
+/// An item in a menu created by a [DropDownButton].
```
Comment (Hixie) [inline packages/flutter/lib/src/material/drop_down.dart]: talk about `T`

File: packages/flutter/lib/src/material/flat_button.dart [hunk 7]
```diff
@@ -34,6 +34,10 @@ import 'theme.dart';
 ///  * [DropDownButton]
 ///  * <https://www.google.com/design/spec/components/buttons.html>
 class FlatButton extends StatelessWidget {
+  /// Creates a flat button.
+  ///
+  /// The [child] argument is required and is typically a [Text] width in all
```
Comment (Hixie) [inline packages/flutter/lib/src/material/flat_button.dart]: strike "width" ?

File: packages/flutter/lib/src/material/material.dart [hunk 8]
```diff
@@ -89,6 +89,8 @@ abstract class InkHighlight {
 
   /// The color use to visually represent the emphasis.
   Color get color;
+
+  /// The color use to visually represent the emphasis.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/material.dart]: This grammar is dubious, but seems to be copy pasta of other bad grammar.

Also, no need to doc the setter.

File: packages/flutter/lib/src/material/material.dart [hunk 9]
```diff
@@ -89,6 +89,8 @@ abstract class InkHighlight {
 
   /// The color use to visually represent the emphasis.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/material.dart]: what emphasis?

also,  grammar bad.

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 10]
```diff
@@ -415,8 +415,20 @@ class _RefreshProgressIndicatorPainter extends _CircularProgressIndicatorPainter
   }
 }
 
-
+/// A progress indicator for refreshing the contents of a widget.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: well the progress indicator itself doesn't refresh the contents... For notifying the user that the contents are being refreshed, maybe?

File: packages/flutter/lib/src/material/raised_button.dart [hunk 11]
```diff
@@ -32,6 +32,10 @@ import 'theme.dart';
 ///  * [FloatingActionButton]
 ///  * <https://www.google.com/design/spec/components/buttons.html>
 class RaisedButton extends StatelessWidget {
+  /// Creates a raised button.
+  ///
+  /// The [child] argument is required and is typically a [Text] width in all
```
Comment (Hixie) [inline packages/flutter/lib/src/material/raised_button.dart]: copy pasta; width -> widget

File: packages/flutter/lib/src/material/toggleable.dart [hunk 12]
```diff
@@ -11,11 +11,15 @@ import 'constants.dart';
 const Duration _kToggleDuration = const Duration(milliseconds: 200);
 final Tween<double> _kRadialReactionRadiusTween = new Tween<double>(begin: 0.0, end: kRadialReactionRadius);
 
-// RenderToggleable is a base class for material style toggleable controls with
-// toggle animations. It handles storing the current value, dispatching
-// ValueChanged on a tap gesture and driving a changed animation. Subclasses are
-// responsible for painting.
+/// RenderToggleable is a base class for material style toggleable controls with
```
Comment (Hixie) [inline packages/flutter/lib/src/material/toggleable.dart]: Remove "RenderToggleable is a".

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/material.dart (hunk 8 above)
Comment (abarth) [inline packages/flutter/lib/src/material/material.dart]: Yeah.

**Pull request merged**